homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,142 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/media_player.firetv/
 """
+import functools
 import logging
-
-import requests
+import os
 import voluptuous as vol
 
 from homeassistant.components.media_player import (
-    PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PLAY,
-    SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF,
-    SUPPORT_TURN_ON, SUPPORT_VOLUME_SET, MediaPlayerDevice)
+    MediaPlayerDevice, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
+    SUPPORT_PLAY, SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE, SUPPORT_STOP,
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_SET, )
 from homeassistant.const import (
-    CONF_DEVICE, CONF_DEVICES, CONF_HOST, CONF_NAME, CONF_PORT, CONF_SSL,
-    STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING, STATE_STANDBY,
-    STATE_UNKNOWN)
+    CONF_HOST, CONF_NAME, CONF_PORT, STATE_IDLE, STATE_OFF, STATE_PAUSED,
+    STATE_PLAYING, STATE_STANDBY)
 import homeassistant.helpers.config_validation as cv
 
+REQUIREMENTS = ['firetv==1.0.7']
+
 _LOGGER = logging.getLogger(__name__)
 
 SUPPORT_FIRETV = SUPPORT_PAUSE | \
     SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PREVIOUS_TRACK | \
-    SUPPORT_NEXT_TRACK | SUPPORT_SELECT_SOURCE | SUPPORT_VOLUME_SET | \
-    SUPPORT_PLAY
+    SUPPORT_NEXT_TRACK | SUPPORT_SELECT_SOURCE | SUPPORT_STOP | \
+    SUPPORT_VOLUME_SET | SUPPORT_PLAY
+
+CONF_ADBKEY = 'adbkey'
+CONF_GET_SOURCE = 'get_source'
+CONF_GET_SOURCES = 'get_sources'
 
-DEFAULT_SSL = False
-DEFAULT_DEVICE = 'default'
-DEFAULT_HOST = 'localhost'
 DEFAULT_NAME = 'Amazon Fire TV'
-DEFAULT_PORT = 5556
-DEVICE_ACTION_URL = '{0}://{1}:{2}/devices/action/{3}/{4}'
-DEVICE_LIST_URL = '{0}://{1}:{2}/devices/list'
-DEVICE_STATE_URL = '{0}://{1}:{2}/devices/state/{3}'
-DEVICE_APPS_URL = '{0}://{1}:{2}/devices/{3}/apps/{4}'
+DEFAULT_PORT = 5555
+DEFAULT_ADBKEY = ''
+DEFAULT_GET_SOURCE = True
+DEFAULT_GET_SOURCES = True
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
-    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_ADBKEY, default=DEFAULT_ADBKEY): cv.string,
```

Comment:
Remove the default and check if files exist.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/80daea67be3aa84a3e511aff54b95d7b3e4e18e7